### ISSUE TITLE
Inaccurate detector's results

### ISSUE THREAD

**Issue by EvdoTheo** [liuyuan-pal/Gen6D#101]
Hello,
I'm trying to evaluate the accuracy of a custom object and I noticed that the detector produces completely inaccurate results. First of all, I need to mention that I need to extract the evaluation metrics ADD-0.1d and Prj-5 for my custom object. The eval.py script requires an align.pkl file, which I extracted from the compute_align_poses.py script. Having completed all the necessary set up steps of the dataset, i run  the eval.py script and i get these results. 
![21-inter](https://github.com/liuyuan-pal/Gen6D/assets/25465584/34a4ae57-4914-4ad6-b04b-4fb2c7f1e77c)
![35-inter](https://github.com/liuyuan-pal/Gen6D/assets/25465584/227114ce-b059-4a9a-ba32-b55592de9096)
 

Moreover, during the prediction process while the initial predictions are not accurate, after a while the refiner outputs better results.
![0](https://github.com/liuyuan-pal/Gen6D/assets/25465584/ea224e85-9d38-4b3e-8014-2349ff47a320)
![40](https://github.com/liuyuan-pal/Gen6D/assets/25465584/a60f0036-47f7-48e1-8bb4-d6fa4a0e25f3)

**Notice** 
The reference images are synthetic images 
![frame15](https://github.com/liuyuan-pal/Gen6D/assets/25465584/0f0a8253-e5db-49c2-ab50-9288609d55b8)

What should i do in your opinion, in order to solve this problem with the detector? Thank you very much for your time.

**Comment by liuyuan-pal** [comment]
Hi, thanks for your interest in our work!
1. Your gt labels look fine!
2. The reason for incorrect `eval.py` is that the scale differences are too large between reference images and query images. You may downsample the query image to a smaller one (and change intrinsics accordingly), then it will be able to produce more accurate results. The reason for that is https://github.com/liuyuan-pal/Gen6D/issues/29#issuecomment-1308379909

**Comment by EvdoTheo** [comment]
Can I tweak the scale difference to check if I'll get better results? And if yes, will this modification decrease the algorithm's performance? Also, where are the intrinsic parameters  located in the code?
Moreover, if I need to train the detector, how should I structure my dataset and what kind of information is required (folders, files, etc)? 

Thank you in advance, I do appreciate it!!

**Comment by liuyuan-pal** [comment]
Yes, you can change the query image size. Changing that to find a suitable size could improve the performance.
If you resize the image here, you should change the `K` accordingly
https://github.com/liuyuan-pal/Gen6D/blob/50aa71b2dbb3198772c025d54d0dd191a918982c/eval.py#L124

The training codes are already included. You may need to read the training codes and find out how to train the detector. In terms of the given object, I think the pretrained detector is OK and you just need to take care of the scale difference in the first frame. Because subsequent frames all are processed by the refiner only.

**Comment by EvdoTheo** [comment]
Great, Thank you for your insights!
